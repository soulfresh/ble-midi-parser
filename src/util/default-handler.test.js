/* eslint-env jest */
jest.mock('./logger');

let mockLog;
let defaultHandler;

beforeEach(() => {
  jest.resetAllMocks();
  mockLog = require('../util/logger').default;
  defaultHandler = require('./default-handler').default;
});

[
  ['mock-one', []],
  ['mock-two', [0, 42]],
  ['mock-three', [true, {}, 'cool']]
].forEach(testData => {
  const [name, args] = testData;
  test(
    `defaultHandler(${JSON.stringify(name)}, ${JSON.stringify(args)})`,
    () => {
      const fn = defaultHandler(name);
      fn(...args);
      expect(mockLog.mock.calls).toMatchSnapshot();
    }
  );
});
