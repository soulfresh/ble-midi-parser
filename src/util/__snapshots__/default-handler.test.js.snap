// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultHandler("mock-one", []) 1`] = `
Array [
  Array [
    "mock-one: []",
  ],
]
`;

exports[`defaultHandler("mock-three", [true,{},"cool"]) 1`] = `
Array [
  Array [
    "mock-three: [true,{},\\"cool\\"]",
  ],
]
`;

exports[`defaultHandler("mock-two", [0,42]) 1`] = `
Array [
  Array [
    "mock-two: [0,42]",
  ],
]
`;
